src/NServiceBus.Transport.Msmq/NonDurable/DetermineMessageDurabilityBehavior.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace NServiceBus.Transport.Msmq
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading.Tasks;
+    using Pipeline;
+
+    class DetermineMessageDurabilityBehavior : IBehavior<IOutgoingLogicalMessageContext, IOutgoingLogicalMessageContext>
```

Comment:
Why can't we do this in the dispatcher? Is it because we don't know the message type in the transport operation? The lack of message type on the operation created for me some friction in SQS, maybe we should consider adding it? 

I am concerned because this API won't work in raw context. Not that I am a big fan of it but...

Reply:
>  Is it because we don't know the message type in the transport operation? The lack of message type on the operation created for me some friction in SQS, maybe we should consider adding it?

yes that is the issue. That might be some addition to be added to the seam.

> I am concerned because this API won't work in raw context. 

yes, this is currently a feature that MSMQ provides to Core and isn't available for raw modes.